CRM/Case/BAO/Case.php

```diff
@@ -1212,9 +1212,20 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
    *
    */
   public static function getRelatedContacts($caseID, $skipDetails = FALSE) {
+    $caseRoles = array();
+    if (!$skipDetails) {
+      $caseInfo = civicrm_api('Case', 'getsingle', array(
```

Comment:
I don't get why there would be an infinite loop. In this call the `return` param is set so there should be no recursion into calling this again.
Also why would `civicrm_api3` trigger a loop but `civicrm_api` would not?